pyrobosim/pyrobosim/gui/main.py

```diff
@@ -318,3 +330,8 @@ def on_collision_polygon_toggle_click(self):
         """Callback to toggle collision polygons."""
         self.canvas.toggle_collision_polygons()
         self.canvas.draw_signal.emit()
+
+    def on_cancel_action_click(self):
+        """Callback to cancel any running action for the current robot."""
+        self.get_current_robot().cancel_actions()
```

Comment:
need to check if current robot is not none -- see the `on_open_click()` and `on_close_click()` functions, for example